crates/core/component/dex/src/component/position_manager/volume_tracker.rs

```diff
@@ -41,24 +41,24 @@ pub(crate) trait PositionVolumeTracker: StateWrite {
             return;
         }
 
-        let trading_pair = new_state.phi.pair.clone();
+        // Get the flows with the first asset being UM, and the other asset)
+        let (flows, other_asset) = {
+            let flows = new_state.flows(&prev_state);
+            let pair = new_state.phi.pair;
+            if pair.asset_1 == *STAKING_TOKEN_ASSET_ID {
+                (flows, pair.asset_2)
+            } else {
+                (flows.flip(), pair.asset_1)
+            }
+        };
 
         // We want to track the **outflow** of staking tokens from the position.
         // This means that we track the amount of staking tokens that have left the position.
         // We do this by comparing the previous and new reserves of the staking token.
         // We **DO NOT** want to track the volume of the other asset denominated in staking tokens.
-        let prev_state = prev_state.as_ref().expect("the previous state exists");
-        let prev_balance = prev_state
-            .reserves_for(*STAKING_TOKEN_ASSET_ID)
-            .expect("the staking token is in the pair");
-
-        let new_balance = new_state
-            .reserves_for(*STAKING_TOKEN_ASSET_ID)
-            .expect("the staking token is in the pair");
-
         // We track the *outflow* of the staking token.
         // "How much inventory has left the position?"
-        let staking_token_outflow = prev_balance.saturating_sub(&new_balance);
+        let staking_token_outflow = flows.lambda_1;
```

Comment:
This is really confusing for me. I think it works because we flip the pair, so in the suspended `Flows` we have made sure that $\Lambda_1$ resolves to the staking token (unsure). But this seems more dangerous than just doing a dumb lookup and saturating sub. And I am still unclear on why we need to do that at all?

I think we should just not track the inflow, keep it tight here, and instead brainstorm a way to uniquely reference position executions so that we can refer to them in various events via some unique identifier.

Reply:
I think having a better means of inspecting position executions for event emission would be great ; we do have an upstream requirement of needing this bit of info though, so I think for now we should keep it.